File: XyzSharp/XyzUtils.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.IO;
using System.IO.Compression;
using System.Security.Cryptography;

namespace XyzSharp
{
    public class XyzUtils
    {
        public static Random random = new Random();

        public static byte[] Deflate(byte[] bytes)
        {
            using (MemoryStream streamUncompressed = new MemoryStream(bytes))
            {
                using (MemoryStream streamCompressed = new MemoryStream())
                {
                    using (DeflateStream compressionStream = new DeflateStream(streamCompressed, CompressionMode.Compress))
                    {
                        streamUncompressed.CopyTo(compressionStream);
                    }

                    return streamCompressed.ToArray();
                }
            }
        }

        public static byte[] Inflate(byte[] bytes)
        {
            using (MemoryStream streamCompressed = new MemoryStream(bytes))
            {
                using (MemoryStream streamDecompressed = new MemoryStream())
                {
                    using (DeflateStream decompressionStream = new DeflateStream(streamCompressed, CompressionMode.Decompress))
                    {
                        decompressionStream.CopyTo(streamDecompressed);
                    }

                    return streamDecompressed.ToArray();
                }
            }
        }

        public static byte[] Encrypt(byte[] message, byte[] key)
        {
            AesManaged aes = new AesManaged();

            byte[] salt = new byte[] { 0x44, 0x12, 0x8, 0x19, 0x4D, 0x51, 0x1B, 0x9 };
            Rfc2898DeriveBytes pdb = new Rfc2898DeriveBytes(key, salt, 2);
            ICryptoTransform encryptor = aes.CreateEncryptor(pdb.GetBytes(32), pdb.GetBytes(16));
            MemoryStream ms = new MemoryStream();
            byte[] encrypted;

            using (CryptoStream cs = new CryptoStream(ms, encryptor, CryptoStreamMode.Write))
            {
                using (BinaryWriter writer = new BinaryWriter(cs))
                    writer.Write(message);
                encrypted = ms.ToArray();
            }

            return encrypted;
        }

        public static byte[] Decrypt(byte[] message, byte[] key)
        {
            AesManaged aes = new AesManaged();

            byte[] salt = new byte[] { 0x44, 0x12, 0x8, 0x19, 0x4D, 0x51, 0x1B, 0x9 };
            Rfc2898DeriveBytes pdb = new Rfc2898DeriveBytes(key, salt, 2);
            ICryptoTransform decryptor = aes.CreateDecryptor(pdb.GetBytes(32), pdb.GetBytes(16));
            MemoryStream ms = new MemoryStream();
            byte[] decrypted;

            using (CryptoStream cs = new CryptoStream(ms, decryptor, CryptoStreamMode.Write))
            {
                using (BinaryWriter writer = new BinaryWriter(cs))
                    writer.Write(message);
                decrypted = ms.ToArray();
            }

            return decrypted;
        }

        public static string MD5(byte[] input)
        {
            MD5 md5 = System.Security.Cryptography.MD5.Create();
            byte[] inputBytes = input;
            byte[] hash = md5.ComputeHash(inputBytes);
            StringBuilder sb = new StringBuilder();
            for (int i = 0; i < hash.Length; i++)
            {
                sb.Append(hash[i].ToString("x2"));
            }
            return sb.ToString().ToLower();
        }

        public static string MD5(string input)
        {
            return MD5(Encoding.UTF8.GetBytes(input));
        }

        public static string SHA256(byte[] input)
        {
            SHA256 sha256 = System.Security.Cryptography.SHA256.Create();
            byte[] inputBytes = input;
            byte[] hash = sha256.ComputeHash(inputBytes);
            StringBuilder sb = new StringBuilder();
            for (int i = 0; i < hash.Length; i++)
            {
                sb.Append(hash[i].ToString("x2"));
            }
            return sb.ToString().ToLower();
        }

        public static string SHA256(string input)
        {
            return SHA256(Encoding.UTF8.GetBytes(input));
        }
    }
}

File: XyzPython/XyzUtils.py
from Crypto.Cipher import AES
from pbkdf2 import PBKDF2
import zlib, hashlib

class InflateError(Exception):
    pass

class DeflateError(Exception):
    pass

class CryptError(Exception):
    pass

class HashError(Exception):
    pass

class NoAddressError(Exception):
    pass

class SendError(Exception):
    pass

def _PKCS7P(data):
    return data+(chr(16-len(data)%16)*(16-len(data)%16)).encode()

def _PKCS7U(data):
    return data[:-data[-1]]

class XyzUtils:
    def deflate(message):
        if type(message) != bytes:
            raise DeflateError("Invalid message data type.")

        return zlib.compress(message)[2:-4]

    def inflate(message):
        if type(message) != bytes:
            raise InflateError("Invalid message data type.")
    
        try:
            _data = zlib.decompress(message, -15)
        except zlib.error:
            raise InflateError("Failed to inflate data.")
        
        return _data
    
    def _getAES(key):
        salt = bytes([0x44, 0x12, 0x08, 0x19, 0x4D, 0x51, 0x1B, 0x09])
        pdb = PBKDF2(key, salt, 2)
        return AES.new(pdb.read(32), AES.MODE_CBC, pdb.read(16))

    def encrypt(message, key):
        if type(message) != bytes:
            raise CryptError("Invalid message data type.")
        if type(key) != bytes:
            raise CryptError("Invalid key data type.")

        aes = XyzUtils._getAES(key)
        return aes.encrypt(_PKCS7P(message))
    
    def decrypt(message, key):
        if type(message) != bytes:
            raise CryptError("Invalid message data type.")
        if type(key) != bytes:
            raise CryptError("Invalid key data type.")

        aes = XyzUtils._getAES(key)
        return _PKCS7U(aes.decrypt(message))

    def md5(message):
        if type(message) != bytes:
            raise HashError("Invalid message data type.")
        return hashlib.md5(message).hexdigest()
    
    def sha256(message):
        if type(message) != bytes:
            raise HashError("Invalid message data type.")
        return hashlib.sha256(message).hexdigest()